source/adapters/level_zero/context.cpp

```diff
@@ -468,7 +468,8 @@ static const uint32_t MaxNumEventsPerPool = [] {
 
 ur_result_t ur_context_handle_t_::getFreeSlotInExistingOrNewPool(
```

Comment:
there's ought to be a better way of allocating an event that doesn't require passing around half a dozen arguments to multiple functions. Maybe some sort of flag?

`auto event = Pool->allocateEvent(HOST_VISIBLE | ENABLE_PROFILER | COUNTER_BASED);`

having functions that accept multiple boolean values is error prone.

Reply:
@pbalcer Thanks for the feedback! We could create a new enum just for this case. However, this seems to be a bit overkill just for this one function. It is only used once event.cpp during event creation.

Reply:
It's not just this function. All the various boolean event parameters are being passed around in multiple functions. Just from a quick search:
`createEventAndAssociateQueue, getEventFromQueueCache, EventCreate, getEventFromContextCache, getFreeSlotInExistingOrNewPool, getZeEventPoolCache`

Reply:
@pbalcer This is quite a large refactoring. Functions with these booleans are abundant. Additionally, it would make sense to add this in **ur_api**, however, we would need to add this in spec before I can do so. Perhaps I can file a new ticket and have this be its own patch?

Reply:
Sure, this was just a suggestion.